### ISSUE TITLE
Git hooks allow to commit failed files by jshint and jscs

### ISSUE THREAD

**Issue by adelura** [ckeditor/ckeditor-boilerplate#12]
It's easy to commit files which are broken. What I did is:
1. `git add`
2. `git commit`
3. Here I've got an error.
4. Work out errors.
5. `git commit`.

What I forgot here is to add new changes to commit, and it let me commit stuff. JSCS and JSHINT should check files which are staged (added to commit).

**Comment by pjasiun** [comment]
> JSCS and JSHINT should check files which are staged (added to commit).

Agree. Another problem I met is checking all changed files. If I change some files, but add to commit only part of them, then all changed files are checked and I am not able to commit files even if files I have added do not have errors.

**Comment by Reinmar** [comment]
+1. I've noticed this too. It's hard to have temporary unstaged files.

**Comment by adelura** [comment]
As far as I can see thery is only first error visible.
